calculator.go

```diff
@@ -1,13 +1,94 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
 // Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+func Add(a, b float64, v ...float64) float64 {
+	r := a + b
+	for _, x := range v {
+		r += x
+	}
+	return r
 }
 
 // Subtract takes two numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+func Subtract(a, b float64, v ...float64) float64 {
+	r := a - b
+	for _, x := range v {
+		r -= x
+	}
+	return r
+}
+
+// Multiply takes two numbers and returns the result of multiplying them together.
+func Multiply(a, b float64, v ...float64) float64 {
+	r := a * b
+	for _, x := range v {
+		r *= x
+	}
+	return r
+}
+
+// Divide takes two numbers and returns the result of dividing the second
+// from the first.
+func Divide(a, b float64, v ...float64) (error, float64) {
+	if b == 0 {
+		return fmt.Errorf("divide-by-zero for Divide(%f, %f)", a, b), 0.0
+	}
+
+	r := a / b
+	for i, x := range v {
+		if x == 0 {
+			return fmt.Errorf("divide-by-zero for Divide(%f, %f), in iteration %v processing variadic float64", r, x, i), 0.0
+		}
+		r /= x
+	}
+	return nil, r
+}
+
+// Sqrt returns the square root of a number
+func Sqrt(a float64) (error, float64) {
+	if a == 0 {
+		return fmt.Errorf("cannot-get-square-root-of-a-negative-number for Sqrt(%f)", a), 0.0
+	}
+
+	return nil, math.Sqrt(a)
+}
+
+func Expression(e string) (error, float64) {
+	var a, b float64
+	var operator string
+
+	strings.ReplaceAll(e, " ", "") // remove all spaces
+
+	t, err := fmt.Sscanf(e, "%f%1s%f", &a, &operator, &b)
+	if err != nil {
+		return fmt.Errorf("Expression() error while parsing %q: %v", e, err), 0.0
+	}
+
+	// t is the number of fields parsed by fmt.Sscanf().
```

Comment:
The fact that we need a comment to explain what `t` is sort of implies that we could use a better name than `t`, doesn't it?